### PR TITLE
Introduce telemetry reporting.

### LINKED ISSUE #2289: Add support for telemetry notifications
- Use https://github.com/redhat-developer/vscode-redhat-telemetry (disabled by default) and follow the listed practices.
- Integrate with [`telemetry/event`](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#telemetry_event) in https://github.com/eclipse/eclipse.jdt.ls/issues/21 .

### PR DESCRIPTION
**Pull request by rgrunber**
- Introduce wrappers for vscode-redhat-telemetry on client side
- Delegate telemetry reported on server side to the wrapper
- Fixes #2289

Signed-off-by: Roland Grunberg <rgrunber@redhat.com>

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -27,7 +28,8 @@
 			"outFiles": [ "${workspaceRoot}/dist/**/*.js" ],
 			"env": {
 				"SERVER_PORT": "3333",
-				"DEBUG_VSCODE_JAVA":"true"
+				"DEBUG_VSCODE_JAVA":"true",
+				"VSCODE_REDHAT_TELEMETRY_DEBUG":"true"
```
Comment (testforstephen) [inline .vscode/launch.json:32]: I'm wondering who will adopt this env?

File: src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		const javaSettings = getNonDefaultJavaSettings(workspace.getConfiguration());
+
+		return telemetryManager.send({
+			name: eventName,
+			properties: { ...data, ...javaSettings }
+		});
+	}
+
+	function sendEmptyStartUp() {
+		if (!serverInitializedReceived) {
```
Comment (testforstephen) [inline src/telemetry.ts:54]: Do we need to check if the telemetry setting is enabled?

File: .vscode/launch.json (hunk 1 above)
Comment (fbricon) [reply .vscode/launch.json:32]: Our telemetry library uses it to display events in the console

File: src/telemetry.ts (hunk 2 above)
Comment (fbricon) [reply src/telemetry.ts:54]: No, this is handled by our telemetry library

File: .vscode/launch.json (hunk 1 above)
Comment (testforstephen) [reply .vscode/launch.json:32]: i see. thanks.

File: package.json [hunk 3]
```diff
@@ -1424,6 +1434,7 @@
   },
```
Comment (fbricon) [inline package.json:1434]: Please add those keys at the root:
```json
    "segmentWriteKey": "Y7Y5Xk8dKEhVZHTmAkFZkqgdN4d7c4lt",
    "segmentWriteKeyDebug": "BflPll7uuKOCm3y0g7JpfXLVBVFBivDE",
```

File: package.json [hunk 4]
```diff
@@ -1424,6 +1434,7 @@
...
   "dependencies": {
     "chokidar": "^3.5.3",
+    "@redhat-developer/vscode-redhat-telemetry": "^0.5.2",
```
Comment (fbricon) [inline package.json:1437]: 0.6.0

**Pull request merged**